lumicks/pylake/kymotracker/kymotrack.py

```diff
@@ -519,21 +524,25 @@ def __len__(self):
         return len(self._src)
 
     def extend(self, other):
-        other = [other] if isinstance(other, KymoTrack) else other
+        if not other:
+            return self
+
+        if not (isinstance(other, KymoTrack) or isinstance(other, self.__class__)):
+            raise TypeError(
+                f"You can only extend a {self.__class__.__name__} with a {self.__class__.__name__} "
+                f"or {KymoTrack.__name__}"
+            )
 
+        other = [other] if isinstance(other, KymoTrack) else other
```

Comment:
looking at this again, I wonder if it's just better to cast to `KymoTrackGroup` rather than `list` up front? Then below after the assertions we can completely get rid of the whole `if` block.
```suggestion
        other = KymoTrackGroup([other]) if isinstance(other, KymoTrack) else other
```

Reply:
Good point. Fixing.